base_layer/wallet/src/transaction_service/protocols/transaction_coinbase_monitoring_protocol.rs

```diff
@@ -154,275 +154,427 @@ where TBackend: TransactionBackend + 'static
-                    completed_tx
-                        .transaction
-                        .body
-                        .outputs()
-                        .iter()
-                        .any(|item| item == &transaction_output);
-            }
-            // If all outputs are present then mark this transaction as mined.
-            if check && !returned_ouputs.is_empty() {
-                self.resources
-                    .output_manager_service
-                    .confirm_transaction(
-                        completed_tx.tx_id,
-                        completed_tx.transaction.body.inputs().clone(),
-                        completed_tx.transaction.body.outputs().clone(),
-                    )
-                    .await
-                    .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
-
-                self.resources
-                    .db
-                    .mine_completed_transaction(completed_tx.tx_id)
-                    .await
-                    .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
-
-                let _ = self
-                    .resources
-                    .event_publisher
-                    .send(Arc::new(TransactionEvent::TransactionMined(completed_tx.tx_id)))
-                    .map_err(|e| {
-                        trace!(
-                            target: LOG_TARGET,
-                            "Error sending event, usually because there are no subscribers: {:?}",
-                            e
-                        );
-                        e
-                    });
+        if !(response.is_synced ||
+            response.location == TxLocation::Mined &&
+                response.confirmations >= self.resources.config.num_confirmations_required as u64)
+        {
+            info!(
+                target: LOG_TARGET,
+                "Base Node reports not being synced, submission will be retried."
+            );
+            return Ok((false, Some(response.height_of_longest_chain)));
+        }
 
+        // Mined?
+        if response.location == TxLocation::Mined {
+            if response.confirmations >= self.resources.config.num_confirmations_required as u64 {
                 info!(
                     target: LOG_TARGET,
-                    "Coinbase Transaction (TxId: {:?}) detected as mined on the Base Layer", completed_tx.tx_id
+                    "Coinbase transaction (TxId: {}) detected as mined and CONFIRMED with {} confirmations",
+                    self.tx_id,
+                    response.confirmations
                 );
-
-                return Ok(BaseNodeResponseType::FetchUtxo(true));
+                return Ok((true, Some(response.height_of_longest_chain)));
             }
+            info!(
+                target: LOG_TARGET,
+                "Coinbase transaction (TxId: {}) detected as mined but UNCONFIRMED with {} confirmations",
```

Comment:
Minor nit: I don't think confirmed or unconfirmed need to be capitalised

Reply:
Kept the log message the same as that for `transaction_broadcast_protocol`